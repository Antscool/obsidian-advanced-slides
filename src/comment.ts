export type Comment = {

	type: string,
	style: string[],
	clazz: string[],
	attributes: Map<string, string>,
}

export class CommentParser {

	private readCommentRegex = /<!--.*-->/;
	private styleRegex = /(?:style="([^"]+)")|(?:style='([^']+)')/;
	private classRegex = /(?:class="([^"]+)")|(?:class='([^']+)')/;
	private parseRegex = /<!--\s*\.(element|slide):\s*(.*)-->/;
	private parsePropertiesRegex = /([^=]*)\s*=\s*"([^"]*)"\s*|([^=]*)\s*=\s*'([^']*)'\s*/g;

	commentToString(comment: Comment): string {
		return `<!-- .${comment.type}: ${this.buildAttributes(comment)} -->`;
	}

	buildAttributes(comment: Comment): string {
		return `style="${comment.style.join('; ')}" class="${comment.clazz.join(' ')}" ${this.attributeMapToString(comment.attributes)}`;
	}

	parseLine(line: string): Comment {
		return this.lineHasComment(line)
			? this.parseComment(this.readCommentStringFromLine(line))
			: null;
	}

	parseComment(comment: string): Comment {
		try {
			let [, type, properties] = this.parseRegex.exec(comment);
			let [style, clazz, attributes] = this.parseProperties(properties);
			return this.buildComment(type, style, clazz, attributes);
		} catch (ex) {
			console.log("ERROR: Cannot parse comment: " + comment);
			return null;
		}
	}

	buildComment(type: string, style: string[] = [], clazz: string[] = [], attributes: Map<string, string> = new Map<string, string>()): Comment {
		return {
			type: type,
			style: style,
			clazz: clazz,
			attributes: attributes,
		}
	}

	private attributeMapToString(attributes: Map<string, string>): string {
		let result = [];
		for (const [key, value] of attributes) {
			result.push(`${key}="${value}"`);
		}
		return result.join(" ");
	}

	private lineHasComment(line: string): boolean {
		return line?.includes("<!--") && line?.includes("-->");
	}

	private readCommentStringFromLine(line: string): string {
		return this.readCommentRegex.exec(line)?.[0] ?? '';
	}

	private parseProperties(properties: string): [string[], string[], Map<string, string>] {
		let result: [string[], string[], Map<string, string>] = [[], [], new Map<string, string>()];

		if (properties.includes('style=')) {
			result[0] = this.parseStyle(properties);
		}

		if (properties.includes('class=')) {
			result[1] = this.parseClass(properties);
		}

		result[2] = this.parseAttributes(properties);

		return result;
	}

	private parseStyle(comment: string): string[] {
		let result = this.styleRegex.exec(comment)?.[1] ?? '';
		return result.split(";").map((attr) => {
			return attr.trim();
		});
	}

	private parseClass(comment: string): string[] {
		let result = this.classRegex.exec(comment)?.[1] ?? '';
		return result.split(" ").map((attr) => {
			return attr.trim();
		});
	}

	private parseAttributes(properties: string): Map<string, string> {
		let m;
		const attributes = new Map<string, string>();

		while ((m = this.parsePropertiesRegex.exec(properties)) !== null) {
			if (m.index === this.parsePropertiesRegex.lastIndex) {
				this.parsePropertiesRegex.lastIndex++;
			}

			let key: string;
			let value: string;

			m.forEach((match, groupIndex) => {
				if (groupIndex == 1 || groupIndex == 3) {
					key = match;
				}
				if (groupIndex == 2 || groupIndex == 4) {
					value = match;
					attributes.set(key, value);
				}
			});

		}

		attributes.delete(undefined);
		attributes.delete('style');
		attributes.delete('class');
		return attributes;
	}

}
