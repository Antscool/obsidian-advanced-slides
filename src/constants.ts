export const ICON_DATA = '<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100"><path fill="currentColor" stroke="currentColor" d="M100 74.242V1.516H0v72.726h45.453v15.153H31.816v9.09h36.368v-9.09H54.547V74.242Zm-90.91-9.09V10.605h81.82v54.547Zm0 0"/><path fill="currentColor" stroke="currentColor" d="M54.547 19.695h9.09V56.06h-9.09ZM72.727 25.758h9.09v30.305h-9.09ZM36.363 31.816h9.09V56.06h-9.09ZM18.184 22.727h9.09v33.335h-9.09Zm0 0"/></svg>';
