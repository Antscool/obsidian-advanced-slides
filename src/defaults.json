{
	"theme": "black",
	"highlightTheme": "zenburn",
	"template": "template/reveal.html",
	"separator": "\r?\n---\r?\n",
	"verticalSeparator": "\r?\n--\r?\n",
	"enableLinks": false,
	"title": "",
	"css": ""
}
