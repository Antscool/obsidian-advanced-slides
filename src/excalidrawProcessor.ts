import { Notice } from "obsidian";
import { ObsidianUtils } from "./obsidianUtils";

export class ExcalidrawProcessor {
	private excalidrawImageRegex = /!\[\[(.*\.excalidraw)\|?([^\]]*)??\]\]\s?(<!--.*-->)?/i

	private utils: ObsidianUtils;

	constructor(utils: ObsidianUtils) {
		this.utils = utils;
	}

	process(markdown: string){
		return markdown
			.split('\n')
			.map((line) => {
				if (this.excalidrawImageRegex.test(line))
					return this.transformLine(line);
				return line;
			})
			.join('\n');
	}

	private transformLine(line: string) {

		var [, image, ext, comment] = this.excalidrawImageRegex.exec(line);
		var imgFile = this.utils.findImageEx(image);

		if (imgFile === null) {
			new Notice(`Cannot find Image for ${image}. Make sure to activate Auto-export SVG/PNG in Excalidraw Settings.`, 8000);
			return line;
		}
		return `![[${imgFile}${ext==undefined ? '' : '|' + ext}]] ${comment ?? ''}`;
	}
}


