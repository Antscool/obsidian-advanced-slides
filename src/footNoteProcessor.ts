
export class FootnoteProcessor {

	private regex = /\[\^([^\]]*)]/gmi;

	process(markdown: string, options: any){

		var output = markdown;

		var result = markdown
        .split(new RegExp(options.separator, 'gmi'))
        .map((slidegroup, index) => {
            return slidegroup
                .split(new RegExp(options.verticalSeparator, 'gmi'))
                .map((slide, index) => {
                    if (this.regex.test(slide)) {
						var newSlide = this.transformFootNotes(slide);
						output = output.replace(slide, newSlide);
                        return newSlide;
                    }
                    return slide;
                })
                .join(options.verticalSeparator);
        })
        .join(options.separator);
    return output;
	}

	transformFootNotes(markdown: string) {

		var input = markdown;
		var noteIdx = 1;
	
		var footNotes = new Map();
	
		var reResult : RegExpExecArray;
		while (reResult = this.regex.exec(input)) {
	
			input = input
				.split('\n')
				.map((line, index) => {
					if (line.includes(reResult[0])) {
						if (line.includes(reResult[0] + ": ")) {
							if (!footNotes.has(reResult[1])) {
								footNotes.set(reResult[1], line.split(reResult[0] + ": ")[1]);
							}
							return "";
						} else {
							var split = line.split(reResult[0]);
	
							var result = split[0].trim();
							result += '<sup id="fnref:' + reResult[1] + '" role="doc-noteref">' + noteIdx + '</sup>';
							result += '\n' + split[1].trim();
	
							noteIdx = noteIdx + 1;
	
							return result;
	
						}
					}
					return line;
				})
				.join('\n');
	
	
		}
	
		var footNotesBlock = '';
		footNotesBlock += '<div class="footnotes" role="doc-endnotes">\n';
		footNotesBlock += '<ol>\n';
	
		footNotes.forEach((value, key) => {
			footNotesBlock += '<li id="fn:' + key + '" role="doc-endnote"><p>' + value + '&nbsp;</p></li>';
		});
	
	
		footNotesBlock += '</ol>\n';
		footNotesBlock += '</div>\n';
	
		return input + '\n' + footNotesBlock;
	}

}


