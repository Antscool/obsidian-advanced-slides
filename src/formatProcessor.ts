
export class FormatProcessor {

	private regex = /==([^=]*)==/gm;
	private commentRegex = /%%([^%]*)%%/gm;

	process(markdown: string){
		return markdown.replaceAll('__','**')
		.replaceAll(this.regex,'<mark>$1</mark>')
		.replaceAll(this.commentRegex,'');
	}

}
