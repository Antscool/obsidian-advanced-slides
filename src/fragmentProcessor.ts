import { CommentParser } from "./comment";
import { Options } from "./options";

export class FragmentProcessor {

	private parser: CommentParser;
	private fragmentCounter = 1;
	private orderedListRegex = /\d\) /g;

	constructor() {
		this.parser = new CommentParser();
	}

	process(markdown: string, options: Options) {
		let output = markdown;

		markdown
			.split(new RegExp(options.separator, 'gmi'))
			.map((slidegroup) => {
				return slidegroup
					.split(new RegExp(options.verticalSeparator, 'gmi'))
					.map((slide) => {
						this.fragmentCounter = 1;

						const newSlide = slide.split('\n')
							.map((line) => {
								if (line.trim().startsWith("+ ") || this.orderedListRegex.test(line.trim())) {
									return this.transformLine(line);
								}
								return line;
							})
							.join('\n');
						output = output.split(slide).join(newSlide);
						return newSlide;

					})
					.join(options.verticalSeparator);
			})
			.join(options.separator);

		return output;
	}

	transformLine(line: string) {
		const comment = this.parser.parseLine(line) ?? this.parser.buildComment('element');

		if (line.includes('<!--')) {
			line = line.substring(0, line.indexOf('<!--'));
		}

		line = line.replaceAll("+ ", "- ");
		line = line.replaceAll(this.orderedListRegex, '1. ');

		if (!comment.attributes.get('data-fragment-index')) {
			comment.attributes.set("data-fragment-index", this.fragmentCounter.toString());
			if (!comment.clazz.includes('fragment')) {
				comment.clazz.push('fragment');
			}
			this.fragmentCounter++;
		}
		const output = line + this.parser.commentToString(comment);
		return output;
	}

}
