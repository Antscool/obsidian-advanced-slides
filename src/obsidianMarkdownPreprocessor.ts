import { App } from "obsidian";
import { BlockProcessor } from "./blockProcessor";
import { FootnoteProcessor } from "./footNoteProcessor";
import { FormatProcessor } from "./formatProcessor";
import { ImageProcessor } from "./imageProcessor";
import { InternalLinkProcessor } from "./internalLinkProcessor";
import { LatexProcessor } from "./latexProcessor";
import { MultipleFileProcessor } from "./multipleFileProcessor";

export class ObsidianMarkdownPreprocessor {

	private multipleFileProcessor : MultipleFileProcessor;
	private blockProcessor : BlockProcessor;
	private imageProcessor : ImageProcessor;
	private internalLinkProcessor : InternalLinkProcessor;
	private footnoteProcessor : FootnoteProcessor;
	private latexProcessor : LatexProcessor;
	private formatProcessor : FormatProcessor;

	constructor(app: App) {
		this.multipleFileProcessor = new MultipleFileProcessor(app);
		this.blockProcessor = new BlockProcessor();
		this.imageProcessor = new ImageProcessor(app);
		this.internalLinkProcessor = new InternalLinkProcessor();
		this.footnoteProcessor = new FootnoteProcessor();
		this.latexProcessor = new LatexProcessor();
		this.formatProcessor = new FormatProcessor();
	}

	process(markdown: string, options: any){
		const afterMultipleFileProcessor = this.multipleFileProcessor.process(markdown);
		const afterBlockProcessor = this.blockProcessor.process(afterMultipleFileProcessor);
		const afterFootNoteProcessor = this.footnoteProcessor.process(afterBlockProcessor, options);
		const afterImageProcessor = this.imageProcessor.process(afterFootNoteProcessor);
		const afterInternalLinkProcessor = this.internalLinkProcessor.process(afterImageProcessor);
		const afterLatexProcessor = this.latexProcessor.process(afterInternalLinkProcessor);
		const afterFormatProcessor = this.formatProcessor.process(afterLatexProcessor);
		return afterFormatProcessor;
	}



}


