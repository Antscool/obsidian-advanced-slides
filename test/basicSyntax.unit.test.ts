import { ObsidianMarkdownPreprocessor } from "src/obsidianMarkdownPreprocessor";
import { when } from "ts-mockito";
import { MockedObsidianUtils, obsidianUtils as utilsInstance} from "./__mocks__/mockObsidianUtils";
import { prepare } from "./testUtils";


test('Basic Markdown Syntax > Headers', () => {

	const input =
		`# This is a heading 1
## This is a heading 2
### This is a heading 3 
#### This is a heading 4
##### This is a heading 5
###### This is a heading 6`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Text style', () => {

	const input =
		`*This text will be italic*

_This will also be italic_

**This text will be bold**

__This will also be bold__

%%This is a Comment(Can't see it)%%

_You **can** combine them_

---

Any word wrapped with two tildes (like ~~this~~) will appear crossed out.

Any word wrapped with two equal signs (like ==this==) will appear as highlighted.`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Lists', () => {

	const input =
		`- Item 1
- Item 2
	- Item 2a
	- Item 2b

---

1. Item 1
1. Item 2
1. Item 3
   1. Item 3a
   1. Item 3b
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Images', () => {

	//TODO: bugfix getAbsolutePath
	when(MockedObsidianUtils.getAbsolutePath("Image.jpg.md")).thenCall( (arg) => {
		return null;
	});

	when(MockedObsidianUtils.getAbsolutePath("Image.jpg|100.md")).thenCall( (arg) => {
		return null;
	});

	when(MockedObsidianUtils.getAbsolutePath('Image.jpg|300x100 <!-- .element: style="object-fit: cover" -->.md')).thenCall( (arg) => {
		return null;
	});

	when(MockedObsidianUtils.findFile('Image.jpg')).thenCall( (arg) => {
		return '/documentation/Image.jpg';
	});

	

	const input =
		`
Insert image with standard markdown syntax

![Image](https://picsum.photos/id/1006/500/300)

---

Insert image that lies outside of your vault

![Image](file:///Users/testUser/Desktop/howToUse.png)

---

Insert image with obsidian markdown syntax

![[Image.jpg]]

---

Scale image to a width of 100 px

![[Image.jpg|100]]

---

Scale image to a width of 300x100 px

![[Image.jpg|300x100]] <!-- .element: style="object-fit: cover" -->

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Links', () => {

	const input =
`External Links

http://obsidian.md - automatic!

[Obsidian](http://obsidian.md)

---

Obsidian URI links

[Link to note](obsidian://open?path=D:%2Fpath%2Fto%2Ffile.md)

[Link to note](obsidian://open?vault=MainVault&file=MyNote.md)

---

Internal Links

This [[Internal link]] will be displayed as normal text

This [[Internal link|Link]] will use its alias for displaying

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Links', () => {

	const input =
`---
enableLinks: true
---

Internal Links

This [[Internal link]] will be displayed as normal text

This [[Internal link|Link]] will use its alias for displaying

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Embeds', () => {

	when(MockedObsidianUtils.getAbsolutePath("Obsidian.md")).thenCall( (arg) => {
		return 'path/to/Obsidian.md';
	});

	when(MockedObsidianUtils.parseFile("path/to/Obsidian.md",'Link')).thenCall( (arg) => {
		return 'Link to Obsidian Homepage: http://obsidian.md';
	});

	when(MockedObsidianUtils.parseFile("path/to/Obsidian.md",null)).thenCall( (arg) => {
		return `# Notes about Obsidian
A knowledge base that works on local Markdown files

# Link
Link to Obsidian Homepage: http://obsidian.md`;
	});

	const input =
`![[Obsidian]]

---

![[Obsidian#Link]]
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Blockquotes', () => {

	const input =
`> Human beings face ever more complex and urgent problems, and their effectiveness in dealing with these problems is a matter that is critical to the stability and continued progress of society.

- Doug Engelbart, 1961
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Inline Code', () => {

	const input =
`### Text inside \`backticks\` on a line will be formatted like code.
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Tables', () => {

	const input =
`First Header | Second Header
------------ | ------------
Content from cell 1 | Content from cell 2
Content in the first column | Content in the second column
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Footnotes', () => {

	const input =
`Here's a simple footnote[^1]

[^1]: meaningful!

`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Math', () => {

	const input =
`$$\begin{vmatrix}a & b\\
c & d
end{vmatrix}=ad-bc$$

You can also do inline math like $s^{-2}_{n}sum_{i=1}^{n}$`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Basic Markdown Syntax > Mermaid', () => {

	const input =
`---
theme: beige
highlightTheme: css/vs2015.css

---

\`\`\`mermaid
sequenceDiagram
    Alice->>+John: Hello John, how are you?
    Alice->>+John: John, can you hear me?
    John-->>-Alice: Hi Alice, I can hear you!
    John-->>-Alice: I feel great!
\`\`\``;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});
