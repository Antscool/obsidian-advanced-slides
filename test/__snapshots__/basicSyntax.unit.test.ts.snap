// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Markdown Syntax > Blockquotes 1`] = `
"> Human beings face ever more complex and urgent problems, and their effectiveness in dealing with these problems is a matter that is critical to the stability and continued progress of society.

- Doug Engelbart, 1961
"
`;

exports[`Basic Markdown Syntax > Embeds 1`] = `
"# Notes about Obsidian
A knowledge base that works on local Markdown files

# Link
Link to Obsidian Homepage: http://obsidian.md

---

Link to Obsidian Homepage: http://obsidian.md
"
`;

exports[`Basic Markdown Syntax > Footnotes 1`] = `
"Here's a simple footnote<sup id=\\"fnref:1\\" role=\\"doc-noteref\\">1</sup>





<div class=\\"footnotes\\" role=\\"doc-endnotes\\">
<ol>
<li id=\\"fn:1\\" role=\\"doc-endnote\\"><p>meaningful!&nbsp;</p></li></ol>
</div>
"
`;

exports[`Basic Markdown Syntax > Headers 1`] = `
"# This is a heading 1
## This is a heading 2
### This is a heading 3 
#### This is a heading 4
##### This is a heading 5
###### This is a heading 6"
`;

exports[`Basic Markdown Syntax > Images 1`] = `
"
Insert image with standard markdown syntax

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"https://picsum.photos/id/1006/500/300\\" alt=\\"Image\\" style=\\"\\" class=\\"\\" ></img></p>


---

Insert image that lies outside of your vault

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/localFileSlash/Users/testUser/Desktop/howToUse.png\\" alt=\\"Image\\" style=\\"\\" class=\\"\\" ></img></p>


---

Insert image with obsidian markdown syntax

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"\\" class=\\"\\" ></img></p>


---

Scale image to a width of 100 px

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"width: 100px\\" class=\\"\\" ></img></p>


---

Scale image to a width of 300x100 px

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"object-fit: cover; width: 300px; height: 100px\\" class=\\"\\" ></img></p>


"
`;

exports[`Basic Markdown Syntax > Inline Code 1`] = `
"### Text inside \`backticks\` on a line will be formatted like code.
"
`;

exports[`Basic Markdown Syntax > Links 1`] = `
"External Links

http://obsidian.md - automatic!

[Obsidian](http://obsidian.md)

---

Obsidian URI links

[Link to note](obsidian://open?path=D:%2Fpath%2Fto%2Ffile.md)

[Link to note](obsidian://open?vault=MainVault&file=MyNote.md)

---

Internal Links

This Internal link will be displayed as normal text

This Link will use its alias for displaying

"
`;

exports[`Basic Markdown Syntax > Links 2`] = `
"

Internal Links

This [Internal link](obsidian://open?vault=test-vault&file=Internal%20link) will be displayed as normal text

This [Link](obsidian://open?vault=test-vault&file=Internal%20link) will use its alias for displaying

"
`;

exports[`Basic Markdown Syntax > Lists 1`] = `
"- Item 1
- Item 2
	- Item 2a
	- Item 2b

---

1. Item 1
1. Item 2
1. Item 3
   1. Item 3a
   1. Item 3b
"
`;

exports[`Basic Markdown Syntax > Math 1`] = `
"\`$$egin{vmatrix}a & b\\\\
c & d
end{vmatrix}=ad-bc$$\`
You can also do inline math like \`$s^{-2}_{n}sum_{i=1}^{n}$\`"
`;

exports[`Basic Markdown Syntax > Mermaid 1`] = `
"


<div class=\\"mermaid\\">
sequenceDiagram
    Alice->>+John: Hello John, how are you?
    Alice->>+John: John, can you hear me?
    John-->>-Alice: Hi Alice, I can hear you!
    John-->>-Alice: I feel great!

</div>
"
`;

exports[`Basic Markdown Syntax > Tables 1`] = `
"First Header | Second Header
------------ | ------------
Content from cell 1 | Content from cell 2
Content in the first column | Content in the second column
"
`;

exports[`Basic Markdown Syntax > Text style 1`] = `
"*This text will be italic*

_This will also be italic_

**This text will be bold**

**This will also be bold**


_You **can** combine them_

---

Any word wrapped with two tildes (like ~~this~~) will appear crossed out.

Any word wrapped with two equal signs (like <mark>this</mark>) will appear as highlighted."
`;
